sdk/attestation/attestation/review/attestation.api.md

```diff
@@ -60,99 +50,61 @@ export interface AttestationClientOptions extends CommonClientOptions {
 }
 
 // @public
-export class AttestationData {
-    constructor(data: Uint8Array, isJson?: boolean);
-    data: Uint8Array;
-    isJson: boolean;
+export interface AttestationPolicyToken extends AttestationToken {
 }
 
 // @public
-export class AttestationResponse<T> {
-    constructor(token: AttestationToken, value: T);
+export interface AttestationResponse<T> {
     token: AttestationToken;
     value: T;
 }
 
 // @public
 export interface AttestationResult {
-    cnf?: any;
-    deprecatedEnclaveHeldData?: Uint8Array;
-    deprecatedEnclaveHeldData2?: Uint8Array;
-    deprecatedIsDebuggable?: boolean;
-    deprecatedMrEnclave?: string;
-    deprecatedMrSigner?: string;
-    deprecatedPolicyHash?: Uint8Array;
-    deprecatedPolicySigner?: JsonWebKey;
-    deprecatedProductId?: number;
-    deprecatedRpData?: string;
-    deprecatedSgxCollateral?: any;
-    deprecatedSvn?: number;
-    deprecatedTee?: string;
-    deprecatedVersion?: string;
     enclaveHeldData?: Uint8Array;
-    exp?: number;
-    iat?: number;
-    inittimeClaims?: any;
+    inittimeClaims: any;
```

Comment:
any chance we could have a better type for any-typed members?

Reply:
The `inittimeClaims` and `runtimeClaims` are caller defined types (the caller provides them as an input to the Attest API, the attestation service validates that they are expressed in the attestation evidence and echoes them into the generated token, which is then sent to a relying party). If you have any suggestions about how this can be expressed, I'd be more than happy to put it in the SDK, but honestly I have no idea how to do that (for Python, the type is Dict, for .Net, the type is "Object").

Other `any` types: 
* policyClaims: This is a set of claims defined by attestation policy documents on the server - similarly, we cannot anticipate the values of this.
* sgxCollateral: This *is* defined for SGX enclaves, but the definition is intentionally fluid - essentially the sgxCollateral is a set of hashes for a half dozen or so inputs to the attestation operation. I can define a model type for this for JS, but it is intentionally left ambiguous (the set of values returned will change over time, and will be different for different types of processors (coffeelake and icelake have different sets of collateral for example).

Reply:
I see. I am fine leaving it as is for now. Generics and `unknown` are the other choices that come to mind. If they're guaranteed to be objects of sorts (as their .NET type imply), maybe you can use `Record<string, unknown>` but no strong preference here.

Reply:
Alternatively, if there is a better object to reflect "An arbitrary JS object", I'm open to switching to that.

Reply:
Btw, to see how this works in practice, look (here)[https://github.com/LarryOsterman/azure-sdk-for-js/blob/48e9cf8617e470923d3d65425c8c6e088711589c/sdk/attestation/attestation/test/public/attestationTests.spec.ts#L266-L281]. The runtimeJson binary string accepted by the service is a UTF8 encoded serialized JSON Web Key, which is emitted as a JSON Web Key in the attestation token.

Reply:
If I switch the type to `unknown`, how do I access it without throwing a TS error?

Reply:
You will have to cast it first to the expected type using `as`.